modules/cherry5-assets-loader/inc/cherry5-assets-loader-handle.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Base lodaer handler class
+ *
+ * @package    Cherry_Framework
+ * @author     Cherry Team <cherryframework@gmail.com>
+ * @copyright  Copyright (c) 2012 - 2017, Cherry Team
+ * @link       http://www.cherryframework.com/
+ * @license    http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+// If this file is called directly, abort.
+if ( ! defined( 'WPINC' ) ) {
+	die;
+}
+
+if ( ! class_exists( 'Cherry5_Assets_Loader_Handle' ) ) {
+
+	/**
+	 * Define Cherry5_Assets_Loader_Handle class
+	 */
+	class Cherry5_Assets_Loader_Handle {
+
+		/**
+		 * Handles list
+		 *
+		 * @var array
+		 */
+		public $handles = array();
+
+		/**
+		 * Handles list
+		 *
+		 * @var array
+		 */
+		public $prepared_handles = array();
+
+		/**
+		 * Handlex context (defined in child classes)
+		 *
+		 * @var string
+		 */
+		public $context = null;
+
+		/**
+		 * Initalize defer loading
+		 *
+		 * @return void
+		 */
+		public function init() {
+
+			if ( null !== $this->context ) {
+				add_filter( $this->context . '_loader_tag', array( $this, 'defer' ), 10, 2 );
```

Comment:
`style_loader_tag`-filter has a 4 params (https://developer.wordpress.org/reference/hooks/style_loader_tag/)
`script_loader_tag`-filter - 3 (https://developer.wordpress.org/reference/hooks/script_loader_tag/)

Maybe add to callback 3-rd param **$src**